src/content/docs/en/reference/integrations-reference.mdx

```diff
@@ -157,6 +157,14 @@ export default {
 }
 ```
 
+<p><Since v="5.x" /></p>
+
+This hook gives you the opportunity to add private config options that are exposed only to integrations.
+There are certain options that are not directly exposed to users, but are allowed to be written from integrations.
+
+For example, the astro config [`serverIslandDynamicBase`](https://github.com/withastro/astro/blob/30d720f519f991720ee0b7c328a4bf4bbf31760d/packages/astro/src/types/public/config.ts#L1979-L1986)
```

Comment:
Heads up that we do not typically link to code outside of docs. Everything in docs should be fully explained so people don't have to leave the page. A full example would have to be provided here.